app/services/MatcherPool.scala

```diff
@@ -65,7 +67,9 @@ class MatcherPool(
   val maxQueuedJobs: Int = 1000,
   val checkStrategy: MatcherPool.CheckStrategy = MatcherPool.blockLevelCheckStrategy,
   val futures: Futures,
-  val checkTimeoutDuration: FiniteDuration = 5 seconds
+  val checkSlowLogDuration: FiniteDuration = 5 seconds,
+  val checkTimeoutDuration: FiniteDuration = 10 seconds,
+  val maybeCloudWatchClient: Option[CloudWatchClient] = None
```

Comment:
Not sold on the idea that we'd always want to have one logging away – in tests for example, which are already busy with logs – but more concretely we'd need to know the stage to instantiate one, and I don't think that's something our `matcherPool` should care about.

Reply:
I was thinking the `matcherPool` wouldn't need to care about stage as it wouldn't instantiate one itself, just be passed a non-optional version. The reason "dryRun" was added was to remove the need for it being optional so it may be more consistent. However I think both ways are fine and I'm happy to leave it.